ext/datadog_profiling_native_extension/heap_recorder.h

```diff
@@ -59,7 +59,7 @@ typedef struct {
 } heap_recorder_iteration_data;
 
 // Initialize a new heap recorder.
-heap_recorder* heap_recorder_new(void);
+heap_recorder* heap_recorder_new(const ddog_prof_ManagedStringStorage *string_storage);
```

Comment:
Minor: Doesn't `const ddog_prof_ManagedStringStorage *` mean the data being pointed to is immutable? :thinking:

Because I'm pretty sure we're mutating it via the libdatadog helpers